tensorflow_datasets/testing/dataset_builder_testing.py

```diff
@@ -386,6 +389,20 @@ def _download_and_prepare_as_dataset(self, builder):
     with self._subTest("config_description"):
       self._test_description_builder_config(builder)
 
+  def _assertNoFileName(self, builder):
+      original_generate_examples = builder._generate_examples
+
+      def new_generate_examples(self, *args, **kwargs):
+        return original_generate_examples(self, *args, **kwargs)
+
+      with mock.patch.object(builder, '_generate_examples', new_generate_examples):
+        keys, data = new_generate_examples(self)
```

Comment:
`new_generate_examples` shouldn't be called explicitly. It will be automatically called when the dataset is generated in `builder.download_and_prepare()`.